spec/ddtrace/contrib/rails/rails_active_job_spec.rb

```diff
@@ -2,77 +2,123 @@
 # in their testing modules.
-require 'sidekiq/testing'
-require 'ddtrace/contrib/sidekiq/server_tracer'
+begin
+  require 'sidekiq/testing'
+  require 'ddtrace/contrib/sidekiq/server_tracer'
+rescue LoadError
+  puts 'Sidekiq testing harness not loaded'
+end
+
 begin
   require 'active_job'
 rescue LoadError
   puts 'ActiveJob not supported in this version of Rails'
 end
 
 require 'ddtrace/contrib/rails/rails_helper'
+require 'ddtrace/contrib/active_job/integration'
 
-RSpec.describe 'Rails with Sidekiq' do
+RSpec.describe 'ActiveJob' do
   before { skip unless defined? ::ActiveJob }
-
   include_context 'Rails test application'
 
-  before do
-    allow(ENV).to receive(:[]).and_call_original
-    allow(ENV).to receive(:[]).with('USE_SIDEKIQ').and_return('true')
-  end
+  context 'with active_job instrumentation' do
+    before do
+      Datadog.configure do |c|
+        c.use :active_job
+      end
 
-  before do
-    Sidekiq.configure_client do |config|
-      config.redis = { url: ENV['REDIS_URL'] }
+      # initialize the application
+      app
     end
 
-    Sidekiq.configure_server do |config|
-      config.redis = { url: ENV['REDIS_URL'] }
-    end
+    after do
+      Datadog.registry[:active_job].reset_configuration!
+      Datadog.configuration.reset!
 
-    Sidekiq::Testing.inline!
-  end
-
-  before { app }
+      # unsubscribe the ActiveSupport notifications
+      ActiveSupport::Notifications.unsubscribe('perform.active_job')
+    end
 
-  context 'with a Sidekiq::Worker' do
     subject(:worker) do
-      stub_const('EmptyWorker', Class.new do
-        include Sidekiq::Worker
-
+      stub_const('EmptyJob', Class.new(ActiveJob::Base) do
         def perform; end
       end)
     end
 
-    it 'has correct Sidekiq span' do
-      worker.perform_async
+    it 'has correct ActiveJob span' do
+      worker.set(queue: :elephants, priority: -10).perform_later
 
-      expect(span.name).to eq('sidekiq.job')
-      expect(span.resource).to eq('EmptyWorker')
-      expect(span.get_tag('sidekiq.job.wrapper')).to be_nil
-      expect(span.get_tag('sidekiq.job.id')).to match(/[0-9a-f]{24}/)
-      expect(span.get_tag('sidekiq.job.retry')).to eq('true')
-      expect(span.get_tag('sidekiq.job.queue')).to eq('default')
+      expect(span.name).to eq('active_job.perform')
+      expect(span.resource).to eq('EmptyJob')
+      expect(span.get_tag('active_job.adapter')).to eq('ActiveJob::QueueAdapters::InlineAdapter')
+      expect(span.get_tag('active_job.job.id')).to match(/[0-9a-f\-]{32}/)
+      expect(span.get_tag('active_job.job.queue')).to eq('elephants')
+
+      if Datadog::Contrib::ActiveJob::Integration.version >= Gem::Version.new('5.0')
+        expect(span.get_tag('active_job.job.priority')).to eq(-10)
+      end
     end
   end
 
-  context 'with an ActiveJob' do
-    subject(:worker) do
-      stub_const('EmptyJob', Class.new(ActiveJob::Base) do
-        def perform; end
-      end)
+  context 'with Sidekiq instrumentation' do
```

Comment:
What happens when Sidekiq+active_job are both enabled? Do we have both Sidekiq and active_job spans in the same trace?

It's not an issue if we have both, but it would be good to have a test case covering that both appear together in the tracer if that's the case.

If not, then no action is needed.